python/cudf/udf_cpp/shim.cu

```diff
@@ -720,3 +722,49 @@ make_definition_corr(BlockCorr, int64, int64_t);
 
 #undef make_definition_corr
 }
+
+/*
+NRT CUDA functions
+*/
+
+// Only used to allocate the right amount of space, see below
+struct meminfo_and_str {
+  NRT_MemInfo mi;
+  udf_string st;
+};
+
+/*
+Create a new MemInfo object holding the reference count of a udf_string. When returning
+new strings, shim functions expect a pointer to a stack allocated buffer into which it
+will construct the udf_string it returns. Since one can not safely build a MemInfo object
+around this stack memory, we store a copy of the udf_string itself next to the MemInfo
+which manages the lifetime of the udf_string.
+*/
+extern "C" __device__ int meminfo_from_new_udf_str(void** nb_retval, void* udf_str)
+{
+  // allocate enough room for both the meminfo and udf_string
+  meminfo_and_str* mi_and_str = (meminfo_and_str*)NRT_Allocate(sizeof(meminfo_and_str));
+  if (mi_and_str != NULL) {
```

Comment:
Should we do something if the allocation fails? Even if we just call `__trap()` so that the kernel dies with an error, it seems like that'd be easier to debug / troubleshoot than proceeding with a null pointer and potentially getting gibberish kernel output.

Reply:
There's a `__trap()` in there now, agree that I'm not sure what the best solution is here. Would the ideal thing be to return an explicit cuda allocation failure to the user here?